### PR TITLE
[SCB-2367] optimize microservice.yaml load log

### PR DESCRIPTION
**Pull request by david6969xin**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/archaius/sources/MicroserviceConfigLoader.java [hunk 1]
```diff
@@ -69,8 +70,8 @@ public void loadAndSort() {
       }
 
       sort();
-    } catch (IOException e) {
-      throw new ServiceCombException("Failed to load microservice config", e);
+    } catch (Exception e) {
+      throw new ServiceCombException("Failed to load microservice configFile " + configFileFromClasspath + " and " + e.getMessage());
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/archaius/sources/MicroserviceConfigLoader.java:74]: throw new ServiceCombException("Failed to load microservice configFile " + configFileFromClasspath, e);

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java [hunk 2]
```diff
@@ -96,14 +96,18 @@ private YAMLUtil() {
 
     for (Map.Entry<String, Object> entry : propertieMap.entrySet()) {
       if (entry.getValue() instanceof Map) {
-        result.putAll(retrieveItems(prefix + entry.getKey(), (Map<String, Object>) entry.getValue()));
+          result.putAll(retrieveItems(prefix + entry.getKey(), (Map<String, Object>) entry.getValue()));
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java:99]: format code

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java [hunk 3]
```diff
@@ -96,14 +96,18 @@ private YAMLUtil() {
...
-          result.put(servicecombKey, entry.getValue());
+        if (entry.getKey() instanceof String){
+          String key = prefix + entry.getKey();
+          if (key.startsWith(CONFIG_CSE_PREFIX)) {
+            String servicecombKey = CONFIG_SERVICECOMB_PREFIX + key.substring(key.indexOf(".") + 1);
+            result.put(servicecombKey, entry.getValue());
+          }
+          result.put(key, entry.getValue());
+        }else {
+          throw new IllegalArgumentException("value config error");
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java:109]: throw new IllegalArgumentException("Find invalid configuration item. Prefix is " + prefix);

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/archaius/sources/MicroserviceConfigLoader.java (hunk 1 above)
Comment (david6969xin) [reply foundations/foundation-config/src/main/java/org/apache/servicecomb/config/archaius/sources/MicroserviceConfigLoader.java]: done

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java (hunk 2 above)
Comment (david6969xin) [reply foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java]: ok

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java (hunk 3 above)
Comment (david6969xin) [reply foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java]: done

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java [hunk 4]
```diff
@@ -55,15 +55,34 @@ private YAMLUtil() {
...
+        } else {
+          flag = true;
+        }
+      } else {
+        flag = false;
+      }
+    }
+    return flag;
+  }
+
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/YAMLUtil.java:85]:
```
  private static boolean isValidMap(Map<Object, Object> data) {
    for (Map.Entry<Object, Object> entry : data.entrySet()) {
      Object key = entry.getKey();
      Object value = entry.getValue();
      if (key instanceof String) {
        if (value instanceof Map) {
          return isValidMap((Map<Object, Object>) value);
        } 
        continue;        
      }
      return false;
    }
   return true;
  }
```

**Review by liubao68 — APPROVED** [review]

**Comment by liubao68** [comment]
retrigger ci

**Pull request merged**